test/backup.ts

```diff
@@ -0,0 +1,516 @@
+      '@google-cloud/common-grpc': {
+        ServiceObject: FakeGrpcServiceObject,
+      },
+      '@google-cloud/promisify': fakePfy,
+    }).Backup;
+  });
+
+  beforeEach(() => {
+    fakeCodec.encode = util.noop;
+    backup = new Backup(INSTANCE, BACKUP_NAME);
+  });
+
+  afterEach(() => sandbox.restore());
+
+  describe('instantiation', () => {
+    it('should promisify all the things', () => {
+      assert(promisified);
+    });
+
+    it('should localize the request function', () => {
+      assert.strictEqual(backup.request, INSTANCE.request);
+    });
+
+    it('should localize the formatted instance name', () => {
+      assert.strictEqual(
+        backup.instanceFormattedName_,
+        INSTANCE.formattedName_
+      );
+    });
+
+    it('should format the name', () => {
+      const formatName_ = Backup.formatName_;
+      const formattedName = 'formatted-name';
+
+      Backup.formatName_ = (instanceName, name) => {
+        Backup.formatName_ = formatName_;
+
+        assert.strictEqual(instanceName, INSTANCE.formattedName_);
+        assert.strictEqual(name, BACKUP_NAME);
+
+        return formattedName;
+      };
+
+      const backup = new Backup(INSTANCE, BACKUP_NAME);
+      assert(backup.formattedName_, formattedName);
+    });
+
+    it('should set the backup id', () => {
+      assert.strictEqual(backup.id, BACKUP_NAME);
+    });
+  });
+
+  describe('create', () => {
+    const INSTANCE_NAME = 'instance-name';
+    const BACKUP_NAME = 'backup-name';
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        parent: INSTANCE_NAME,
+        backupId: BACKUP_NAME,
+        backup: {
+          name: BACKUP_FORMATTED_NAME,
+          database: DATABASE_FORMATTED_NAME,
+          expireTime: BACKUP_EXPIRE_TIME.toStruct(),
+        },
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'createBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.create({
+        databasePath: DATABASE_FORMATTED_NAME,
+        expireTime: BACKUP_EXPIRE_TIME,
+      });
+    });
+
+    it('should accept gaxOptions and a callback', async () => {
+      const gaxOptions = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, gaxOptions);
+      };
+
+      await backup.create(
+        {
+          databasePath: DATABASE_FORMATTED_NAME,
+          expireTime: BACKUP_EXPIRE_TIME,
+          gaxOptions,
+        },
+        assert.ifError
+      );
+    });
+
+    describe('error', () => {
+      const REQUEST_RESPONSE_ARGS = [new Error('Error.'), null, null, null];
```

Comment:
This should be in the format of "error, operation, apiResponse". So let's simulate an error response as "error = new Error('Error.'), operation = null, apiResponse = {}".

Then, confirm we get those exact arguments in the user-provided callback.